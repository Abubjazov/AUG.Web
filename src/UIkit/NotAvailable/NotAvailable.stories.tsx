import type { Meta, StoryObj } from '@storybook/react'

import NotAvailable, { NotAvailableProps } from './NotAvailable'

import '/src/index.css'

const meta: Meta<NotAvailableProps> = {
  component: NotAvailable,
  title: 'UIkit/Not Available',
  parameters: {
    docs: {
      description: {
        component: 'Standard interface component - NotAvailable',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    text: {
      description: 'Title text',
    },
  },
}

export default meta

type Story = StoryObj<NotAvailableProps>

export const Default: Story = {
  args: { text: 'Title text' },
  decorators: [Story => <Story />],
}
