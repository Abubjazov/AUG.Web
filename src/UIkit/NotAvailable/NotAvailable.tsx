import { FC } from 'react'

import styles from './NotAvailable.module.css'

export interface NotAvailableProps {
  text: string
}

const NotAvailable: FC<NotAvailableProps> = ({ text }) => {
  return (
    <div className={styles.fallback}>
      <div className={styles['stars']}>
        <div className={styles['central-body']}>
          <span className={styles.text}>{text}</span>
          <div className={styles['objects']}>
            <img
              className={styles['object-rocket']}
              src="images/rocket.svg"
              width="40px"
            />
            <div className={styles['earth-moon']}>
              <img
                className={styles['object-earth']}
                src="images/earth.svg"
                width="100px"
              />
              <img
                className={styles['object-moon']}
                src="images/moon.svg"
                width="80px"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

export default NotAvailable
