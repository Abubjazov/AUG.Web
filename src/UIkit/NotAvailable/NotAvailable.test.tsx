import { render } from '@testing-library/react'

import NotAvailable from './NotAvailable'

describe('NotAvailable', () => {
  test('should render NotAvailable default', () => {
    const { asFragment } = render(<NotAvailable text={'Not available'} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
