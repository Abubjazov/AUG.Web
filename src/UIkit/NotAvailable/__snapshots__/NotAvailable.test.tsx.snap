// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotAvailable > should render NotAvailable default 1`] = `
<DocumentFragment>
  <div
    class="_fallback_32ff0f"
  >
    <div
      class="_stars_32ff0f"
    >
      <div
        class="_central-body_32ff0f"
      >
        <span
          class="_text_32ff0f"
        >
          Not available
        </span>
        <div
          class="_objects_32ff0f"
        >
          <img
            class="_object-rocket_32ff0f"
            src="images/rocket.svg"
            width="40px"
          />
          <div
            class="_earth-moon_32ff0f"
          >
            <img
              class="_object-earth_32ff0f"
              src="images/earth.svg"
              width="100px"
            />
            <img
              class="_object-moon_32ff0f"
              src="images/moon.svg"
              width="80px"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
